### PR TITLE
MF-136 - Setup staging and test environments

### LINKED ISSUE #136: Setup staging and test environments
As an operator I would expect an automated deployment facilities to be present. All technology choices should be elaborated as part of the contribution. Automated deployment procedure should be verified on our staging and test environments.

### PR DESCRIPTION
**Pull request by dborovcanin**
This pull request resolves #136.

Setup Mainflux services to work in Kubernetes environment.

### CODE REVIEW

**Review by drasko — COMMENTED** [review]
LGTM

File: deployment/kubernetes/ingress-controller/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: load-balancer
+  annotations:
+    # ingress.kubernetes.io/rewrite-target: /api
```
Comment (mijicd) [inline deployment/kubernetes/ingress-controller/ingress.yaml]: Same here.

File: deployment/kubernetes/nomralizer/deployment.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+apiVersion: apps/v1beta1
```
Comment (mijicd) [inline deployment/kubernetes/nomralizer/deployment.yml]: Fix directory name typo.

**Review by mijicd — CHANGES_REQUESTED** [review]
Flatten the directory structure by moving the `deployment/kubernetes` contents into `k8s` directory in project root.

File: k8s/nginx/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# NGINX as Reverse Proxy for Mainflux services
```
Comment (mijicd) [inline k8s/nginx/README.md]: As far as i can see, this is already a part of `k8s/README.md`. Hence, please remove it.

**Review by mijicd — CHANGES_REQUESTED** [review]

**Review by mijicd — APPROVED** [review]

**Pull request merged**